app/src/main/java/dev/shreyaspatil/foodium/ui/details/PostDetailsActivity.kt

```diff
@@ -76,6 +89,22 @@ class PostDetailsActivity : BaseActivity<PostDetailsViewModel, ActivityPostDetai
                 supportFinishAfterTransition()
                 return true
             }
+
+            R.id.action_share -> {
+                if (getPost != null) {
+                    val post = getPost!!
+                    val shareMsg = """
+                        "${post.title}" by ${post.author} on Foodium App
+                    """.trimIndent()
+
+                    val intent = Intent()
+                    intent.action = Intent.ACTION_SEND
+                    intent.putExtra(Intent.EXTRA_TEXT, shareMsg)
+                    intent.type = "text/plain"
+                    startActivity(Intent.createChooser(intent, "Share to:"))
```

Comment:
You can use `ShareCompat` class which is recommended by Android officials for such use case. Here's the reference: https://medium.com/androiddevelopers/sharing-content-between-android-apps-2e6db9d1368b